server/channels/app/post_file_change_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProcessPostFileChanges(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("no files", func(t *testing.T) {
+		oldPost := &model.Post{FileIds: []string{}}
+		newPost := &model.Post{FileIds: []string{}}
+
+		appErr := th.App.processPostFileChanges(th.Context, newPost, oldPost)
+		require.Nil(t, appErr)
+		require.Equal(t, 0, len(newPost.FileIds))
+	})
+
+	t.Run("have files but nothing changed", func(t *testing.T) {
+		oldPost := &model.Post{FileIds: []string{"file_id_1", "file_id_2"}}
+		newPost := &model.Post{FileIds: []string{"file_id_1", "file_id_2"}}
+
+		appErr := th.App.processPostFileChanges(th.Context, newPost, oldPost)
+		require.Nil(t, appErr)
+		require.Equal(t, 2, len(newPost.FileIds))
```

Comment:
Can you share what would it help with?

Reply:
It may be overkill. The only reason to do it is to verify not only the length is the same, but the same values.

Trying to think of a bug that this would not catch and such test would catch, I only find very fabricated examples, so I am happy leaving it like this.